### PR TITLE
#1041 localization incomplete

### LINKED ISSUE #1040: ByProgramModal tabs should be capitalized
## Describe the bug
The tabs on `ByProgramModal` - `general x` and `program x` are not capitalized.

### To reproduce
1. Go to `ByProgramModal`
2. See monstrosity 

### Expected behaviour
Should be capitalized

### Version and device info
 - App version: 2.3.4
 - Tablet model: All
 - OS version: All

### Additional context
Caught in build testing here: https://github.com/openmsupply/mobile/issues/815#issuecomment-507079999

### LINKED ISSUE #1041: French localization is incomplete
## Describe the bug
As seen here https://github.com/openmsupply/mobile/issues/889#issuecomment-507093202 - french translations are not being used and/or are inconsistent.

### To reproduce
1. Go to Language modal, select french
2. Click on create requisition [With a program defined]
3. See error

### Expected behaviour
To be translated correctly

### Version and device info
 - App version: 2.3.4

### Additional context
Translations are there, localizations have been put in place, but I have done it incorrectly. Should also check the consistency in the translations, i.e. capitalization and fit correctly etc.

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1041 Fixes #1040 

Bunch of fixes for localization. I factored out columns in a few places and tried to have localization as constants outside of components. This was a mistake and caused translations not to actually be used/changed. Making this a function causes the translations to be used as they are evaluated when the component is rendered

 Also, `programStrings` was not put in the `setCurrentLanguage` function.

## Change summary
- Added a bunch of correct capitilization to strings
- Fixed few spelling mistakes/duplication/choices
- Added the use of some translations
- Added a fix for translations actually being in use in the pages where I refactored during IC changes and broke everything. Specifically: `SupplierRequisitionPage` and `ByProgramModal`
- Tried to accommodate all languages in the spacing within column headers
- Put `programStrings` in the `setCurrentLanguage` function

## Testing
Steps to reproduce or otherwise test the changes of this PR:
- [ ] Navigating throughout the app in French language is Frenched up
- [ ] Sufficient spacing for English and French languages everywhere
- [ ] Specifically the `ByProgramModal` and `SupplierRequisitionPage` components use localization [**Note: `programStrings` only has English and French translations**]

### Related areas to think about
I have checked other languages also, but could have missed something in those also, which could be made better.

#### Note: Should be going into master to release in a patch for IC

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/design-discussion.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+name: Design discussion
```
Comment (Chris-Petty) [inline .github/ISSUE_TEMPLATE/design-discussion.md]: Not sure why, but you've got a folder `github/` rather than the hidden `.github/`

Comment (Chris-Petty) [reply .github/ISSUE_TEMPLATE/design-discussion.md]: Huh, at least it looked like that in the review window. Still not sure why you're getting this in your PR.

File: src/localization/tableStrings.js [hunk 2]
```diff
@@ -55,56 +55,60 @@ export const tableStrings = new LocalizedStrings({
...
-    actual_quantity: 'Quantité\nRéelle',
-    available_stock: 'Quantité\ndisponible',
-    batch_expiry: 'Expiration',
-    batch: 'Lot',
-    category: 'Catégorie',
+    actual_quantity: 'QUANTITÉ\nRÉELLE',
+    available_stock: 'QUANTITÉ\nDISPONIBLE',
+    batch_expiry: 'EXPIRATION',
+    batch: 'LOT',
+    category: 'catégorie',
```
Comment (mark-prins) [inline src/localization/tableStrings.js]: this feel nit-picky.. can it be an uppercase 'C' please?

File: src/localization/tableStrings.js [hunk 3]
```diff
@@ -55,56 +55,60 @@ export const tableStrings = new LocalizedStrings({
...
     created_date: 'DATE CRÉEE',
     current_stock: 'STOCK ACTUEL',
     customer: 'CLIENT',
     delete: 'SUPPRIMER',
-    department: 'Départment',
-    difference: 'Différence',
-    earliest_expiry: "Date d'exipration la plus proche",
-    entered_date: "Date d'entrée",
-    finalised: 'Finalisé',
+    department: 'départment',
```
Comment (mark-prins) [inline src/localization/tableStrings.js]: :nit uppercase on the 'D'?

File: src/localization/tableStrings.js (hunk 2 above)
Comment (joshxg) [reply src/localization/tableStrings.js]: 058f7a33b0b60f53970cabd64e80354d79541574

File: src/localization/tableStrings.js (hunk 3 above)
Comment (joshxg) [reply src/localization/tableStrings.js]: 058f7a33b0b60f53970cabd64e80354d79541574 thanks!

**Pull request merged**